core/config/value.go

```diff
@@ -479,23 +480,34 @@ func (cv Value) getValueStruct(key string, target interface{}) (interface{}, boo
 
 					item := destSlice.Index(ai)
 					item.Set(reflect.ValueOf(itemValue))
-					found = true
 				} else {
 					break
 				}
 			}
 			if destSlice.Len() > 0 {
 				fieldValue.Set(destSlice)
 			}
-		}
-
-	}
+		case bucketMap:
+			destMap := reflect.MakeMap(fieldType).Interface()
+			val := global.GetValue(childKey).Value()
 
-	if found {
-		if errs := validator.Validate(target); errs != nil {
-			return nil, true, errs
+			if val != nil {
+				switch v := val.(type) {
+				case map[interface{}]interface{}:
```

Comment:
Changed to an if statement. yamlnode is always made as map[interface{}]interface{}, this is a check for that. I'll add a comment.